CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+# Contributing
+
+This document provides guidelines for contributing to this repository.
+
+## Development Environment Setup
+
+### Prerequisites
+* Basic tools: `curl`, `jq`, `oc`
+* go: v1.19
+* Container engine: `podman`
+
+## Running a test environment
+
+### Kwok Container
+"Kwok" is a Kubernetes SIGs-hosted project. KWOK is an abbreviation for Kubernetes Without Kubelet. Kwok simply simulates the node's behaviour. As a result, it can mimic a high number of nodes and pods while consuming only a small amount of memory.
+
+To run the Kwok container with the Kwok Kubernetes tool, follow these steps:
+
+1. Bring up the Kwok cluster by running the following command:
+
+```
+podman run --rm -d -p 8080:8080 --name kwok-cluster registry.k8s.io/kwok/cluster:v0.3.0-k8s.v1.27.3
+```
+
+2. Once the Kwok cluster is up and running, set the cluster details in the OpenShift client with the following command:
+
+```
+oc config set-cluster kwok --server=http://127.0.0.1:8080
+```
+
+3. Create a new context (you only need to set the cluster once) for the Kwok cluster with the following command:
+
+```
+oc config set-context kwok --cluster=kwok
+```
+4. Set the Kwok context as the current context, if you've previously switched to another cluster, with the following command:
+```
+oc config use-context kwok
+```
+
+Now you can access the cluster using kubectl, e.g.: `kubectl get ns`.
+
+### Setting up a containerized Splunk instance
```

Comment:
Likewise, this should be updated to build and run our image rather then a generic one.

Reply:
@ifireball should this section use podman play kube or with podman build and run? what the preference here.

Reply:
whatever makes the life of the reader easier

Reply:
Done